java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/ArrowFlightConnection.java

```diff
@@ -158,6 +158,7 @@ ArrowFlightSqlClientHandler getClientHandler() {
   synchronized ExecutorService getExecutorService() {
     return executorService =
         executorService == null
+        // Refactoring this would require defining a new custom thread factory class
```

Comment:
Naming scheme of io.netty.util.concurrent.DefaultThreadFactory: "<poolName>-<poolID>-<threadID>" (name is explicitly defined [here](https://github.com/netty/netty/blob/29bab2e58cca629939cd92a7ccc0bd08ec1c2d7f/common/src/main/java/io/netty/util/concurrent/DefaultThreadFactory.java#L105), with `prefix` defined [here](https://github.com/netty/netty/blob/29bab2e58cca629939cd92a7ccc0bd08ec1c2d7f/common/src/main/java/io/netty/util/concurrent/DefaultThreadFactory.java#L93))
Would need to define an AtomicInteger (similar to [this one](https://github.com/netty/netty/blob/29bab2e58cca629939cd92a7ccc0bd08ec1c2d7f/common/src/main/java/io/netty/util/concurrent/DefaultThreadFactory.java#L31)) to keep track of how many pools have been created.